packages/wrangler/src/containers/registries.ts

```diff
@@ -0,0 +1,332 @@
+import {
+	cancel,
+	endSection,
+	log,
+	startSection,
+	updateStatus,
+} from "@cloudflare/cli";
+import {
+	ApiError,
+	getAndValidateRegistryType,
+	ImageRegistriesService,
+} from "@cloudflare/containers-shared";
+import { ExternalRegistryKind } from "@cloudflare/containers-shared/src/client/models/ExternalRegistryKind";
+import { APIError, UserError } from "@cloudflare/workers-utils";
+import { handleFailure, promiseSpinner } from "../cloudchamber/common";
+import { confirm, prompt } from "../dialogs";
+import { isNonInteractiveOrCI } from "../is-interactive";
+import { logger } from "../logger";
+import { createSecret, createStore, listStores } from "../secrets-store/client";
+import { validateSecretName } from "../secrets-store/commands";
+import { getAccountId } from "../user";
+import { readFromStdin, trimTrailingWhitespace } from "../utils/std";
+import { formatError } from "./deploy";
+import { containersScope } from ".";
+import type {
+	CommonYargsArgv,
+	StrictYargsOptionsToInterface,
+} from "../yargs-types";
+import type { Config } from "@cloudflare/workers-utils";
+
+export const registryCommands = (yargs: CommonYargsArgv) => {
+	return yargs
+		.command(
+			"configure <DOMAIN>",
+			"Configure credentials for a non-Cloudflare container registry",
+			(args) => registryConfigureYargs(args),
+			(args) =>
+				handleFailure(
+					`wrangler containers registries configure`,
+					registryConfigureCommand,
+					containersScope
+				)(args)
+		)
+		.command(
+			"list",
+			"List all configured container registries",
+			(args) => registryListYargs(args),
+			(args) =>
+				handleFailure(
+					`wrangler containers registries list`,
+					registryListCommand,
+					containersScope
+				)(args)
+		)
+		.command(
+			"delete <DOMAIN>",
+			"Delete a configured container registry",
+			(args) => registryDeleteYargs(args),
+			(args) =>
+				handleFailure(
+					`wrangler containers registries delete`,
+					registryDeleteCommand,
+					containersScope
+				)(args)
+		);
+};
+function registryConfigureYargs(args: CommonYargsArgv) {
+	return (
+		args
+			.positional("DOMAIN", {
+				describe: "Domain to configure for the registry",
+				type: "string",
+				demandOption: true,
+			})
+			.option("public-credential", {
+				type: "string",
+				description:
+					"The public part of the registry credentials, e.g. `AWS_ACCESS_KEY_ID` for ECR",
+				demandOption: true,
+				alias: ["id", "aws-access-key-id"],
```

Comment:
```suggestion
				alias: ["aws-access-key-id"],
```
might risk confusion with secrets store id? I don't have a strong opinion here though